### ISSUE TITLE
Crash with Catalyst

### ISSUE THREAD

**Issue by rrenna** [simonbs/Runestone#40]
**Describe the bug**
When adding a TextView as a subview in a UIViewController in a Catalyst build there's an immediate crash when tapping into the TextView (and making first responder).

Error is:
```Runestone/TextInputView.swift:1287: Fatal error: Positions must be of type IndexedPosition``` looks like `position` is nil.

**Screenshots**
<img width="659" alt="image" src="https://user-images.githubusercontent.com/849462/167609955-8ee2b347-5000-4eda-ac6b-c32df440d5b7.png">

**Comment by simonbs** [comment]
@rrenna Thanks for reporting the issue.

Runestone isn't currently _officially_ supported on Catalyst. However, I've done some initial testing and ran into the same issue as you. It boggles my mind how and why UIKit can passes an instance of UITextPosition to `compare(_:to:)` that isn't an instance of our IndexPosition.

In my testing I simply returned `.same` for any other instance of UITextPosition than those of IndexPosition. So like:

```swift
func compare(_ position: UITextPosition, to other: UITextPosition) -> ComparisonResult {
    guard let indexedPosition = position as? IndexedPosition, let otherIndexedPosition = other as? IndexedPosition else {
        return .same
    }
    if indexedPosition.index < otherIndexedPosition.index {
        return .orderedAscending
    } else if indexedPosition.index > otherIndexedPosition.index {
        return .orderedDescending
    } else {
        return .orderedSame
    }
}
```

I do not have any immediate plans to make this change to Runestone myself since I'm sure that opens up for other optimisations that should be done on Catalyst, and right now I don't have the capacity to support Catalyst. PRs to improve the support for Catalyst are very welcome though 😊

**Comment by simonbs** [comment]
This should be fixed in #48.

**Comment by rrenna** [comment]
Thank you for addressing this despite its low priority. Very much  appreciated.
